karate-core/src/main/java/org/glassfish/jersey/karate/uri/internal/CharacterIterator.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2012, 2018 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
```

Comment:
yikes, I now need to check if this mixing of EPL licensed code is ok from various angles. do you have any experience with this :(

Reply:
Haha yep, thats a different licence from original which is also CDDL. But perhaps we dont know enough about it so we can just use a shaded version to make it simpler?

Reply:
yes sir, lets go with the shaded option ! you can use the existing pom as a reference

Reply:
Done. With so many conflicting commits it might be worth squashing them during the merge.